### ISSUE TITLE
ERROR with volume long syntax

### ISSUE THREAD

**Issue by rdxmb** [docker/docker.github.io#4615]
File: [compose/compose-file/index.md](https://docs.docker.com/compose/compose-file/), CC @gbarr01

Just copied the example of the volume long syntax in https://docs.docker.com/compose/compose-file/#long-syntax-3 into my docker-compose.yml and got:

```
root@docker-single-host:/tmp# docker-compose up
ERROR: In file './docker-compose.yml', volume must be a mapping, not an array.
root@docker-single-host:/tmp# cat docker-compose.yml 
version: "3.2"
services:
  web:
    image: nginx:alpine
    ports:
      - "80:80"

networks:
  webnet:

volumes:
  - type: volume
    source: mydata
    target: /data
    volume:
      nocopy: true
  - type: bind
    source: ./static
    target: /opt/app/static
root@docker-single-host:/tmp# docker-compose --version
docker-compose version 1.16.1, build 6d1ac21
```

//edited:
As the following code works, is this the example it should be?

```
version: "3.2"
services:
  web:
    image: nginx:alpine
    ports:
      - "80:80"
    volumes:
      - type: volume
        source: mydata
        target: /data
        volume:
          nocopy: true
      - type: bind
        source: ./static
        target: /opt/app/static

networks:
  webnet:

volumes:
  mydata:
```

**Comment by mstanleyjones** [comment]
I think this might be fixed by #4491. Can you please take a look?

**Comment by gbarr01** [comment]
Hi, @mstanleyjones, almost. Both #4491 and #4615 are correct but I added top-level entries (from this issue) to expand the sample and give context.

**Comment by rdxmb** [comment]
ok, when should I close this issue? With the pull request merged?

**Comment by gbarr01** [comment]
Fixed per #4491. Thanks for filing it.
